services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/switchmanager/SwitchManager.java

```diff
@@ -1593,4 +1606,36 @@ private MacAddress getPortHwAddress(IOFSwitch sw, int portNumber) throws SwitchO
     private boolean isDefaultRule(long cookie) {
         return (cookie & DEFAULT_RULES_MASK) != 0L;
     }
+
+    private boolean meterExists(IOFSwitch sw, long meter, long rate) {
+        try {
+            verifySwitchSupportsMeters(sw);
+            OFMeterConfig meterConfig = getMeter(sw.getId(), meter);
+            long expectedRate = isSwitchSupportsPktpsFlag(sw) ? rate : getRateForMeterInKbps(rate);
+
+            List<OFMeterBand> ofMeterBands = meterConfig.getEntries();
+            if (ofMeterBands == null || ofMeterBands.isEmpty()) {
+                return false;
+            }
+
+            OFMeterBand meterBand = ofMeterBands.get(0);
```

Comment:
Danger assumption... The method name tells about check for meter existence, but method itself not only check the existence, but also compare meter options. It will lead to problems - this method report "meter does not exist" and caller call setup meter with appropriate options, but get error from switch "meter ID already exists". Confusing?